src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -829,6 +829,49 @@ public static void DeserializeUserDefinedDictionaryThrows()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<IImmutableDictionaryWrapper>(json));
         }
 
+        [Fact]
+        public static void Regression38643_Serialize()
+        {
+            // Arrange
+            var value = new Regression38643_Parent()
+            {
+                Child = new Dictionary<string, Regression38643_Child>()
+                {
+                    ["1"] = new Regression38643_Child()
+                    {
+                        A = "1",
+                        B = string.Empty,
+                        C = Array.Empty<string>(),
+                        D = Array.Empty<string>(),
+                        F = Array.Empty<string>(),
+                        K = Array.Empty<string>(),
+                    }
+                }
+            };
+
+            var actual = JsonSerializer.ToString(value, new JsonSerializerOptions() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+
+            // Assert
+            Assert.NotNull(actual);
+            Assert.NotEmpty(actual);
+        }
+
+        [Fact]
+        public static void Regression38643_Deserialize()
+        {
+            // Arrange
+            string json = "{\"child\":{\"1\":{\"a\":\"1\",\"b\":\"\",\"c\":[],\"d\":[],\"e\":null,\"f\":[],\"g\":null,\"h\":null,\"i\":null,\"j\":null,\"k\":[]}}}";
```

Comment:
What happens if the `c` in `child` was upper case? We should consider adding such a test as well where camel case naming policy results in a property not found (maybe even capitalizing everything like in the original issue).

If we don't yet have a test for this, we should also add a simple test with a few properties (for both serialize/deserialize) where enabling camel case causing name conflicts (duplicate properties).

Reply:
then the `Child` property isn't set.